thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobRunner.java

```diff
@@ -72,32 +70,45 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
     } else {
       detectionJobContext.setAnomalyFunctionSpec(anomalyFunctionSpec);
 
+      DateTime monitoringWindowStartTime = detectionJobContext.getWindowStartTime();
+      DateTime monitoringWindowEndTime = detectionJobContext.getWindowEndTime();
+      boolean needCleanUpMonitoringWindow = (monitoringWindowEndTime == null);
```

Comment:
I thought about clean up start and end window all the time, but I decide to make this function does not change the state of detectionJobContext in order to prevent future problems, e.g., an adhoc detection job needs the information of the aligned start and end window after the job is executed.